### PR TITLE
GH Actions: run CI on pushes to master, and PRs that target master

### PR DESCRIPTION
**Pull request by warner**
refs #466

### CODE REVIEW

**Comment by warner** [comment]
I copied this config from the eslint repo. @katelynsills I know you've got more experience than I do in creating PRs that target non-trunk branches.. would you prefer we run CI on all of those PRs too?

File: .github/workflows/test-all-packages.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: Test all Packages
 
-on: [pull_request]
+on:
+ push:
+   branches: [master]
+ pull_request:
```
Comment (katelynsills) [inline .github/workflows/test-all-packages.yml:9]: I think we want to have pull_request on every branch, not just master

**Review by katelynsills — CHANGES_REQUESTED** [review]

**Comment by katelynsills** [comment]
> I copied this config from the eslint repo. @katelynsills I know you've got more experience than I do in creating PRs that target non-trunk branches.. would you prefer we run CI on all of those PRs too?

Whoops, sorry, just saw this comment. Yeah, if we could run CI on all the PRs, that'd be great!

File: .github/workflows/test-all-packages.yml (hunk 1 above)
Comment (warner) [reply .github/workflows/test-all-packages.yml:9]: ok let's see if that change has the right syntax

**Review by katelynsills — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**